packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/modules/internals/DevMenuInternalMenuControllerModule.kt

```diff
@@ -84,4 +90,85 @@ class DevMenuInternalMenuControllerModule(private val reactContext: ReactContext
       promise.resolve(result)
     }
   }
+
+  override fun getDevSettingsAsync(promise: Promise) {
+    val reactInstanceManager = devMenuManger.getSession()?.reactInstanceManager
+    val map = Arguments.createMap()
+
+    if (reactInstanceManager != null) {
+      val devDelegate = DevMenuDevToolsDelegate(devMenuManger, reactInstanceManager)
+      val devSettings = devDelegate.devSettings
+      val devInternalSettings = (devSettings as? DevInternalSettings)
+
+      map.apply {
+        if (devInternalSettings != null) {
+          putBoolean("isDebuggingRemotely", devSettings.isRemoteJSDebugEnabled)
+          putBoolean("isElementInspectorShown", devSettings.isElementInspectorEnabled)
+          putBoolean("isHotLoadingEnabled", devSettings.isHotModuleReplacementEnabled)
+          putBoolean("isPerfMonitorShown", devSettings.isFpsDebugEnabled)
+        }
+      }
+    }
+
+    promise.resolve(map)
+  }
+
+  override fun getBuildInfoAsync(promise: Promise) {
+    val map = Arguments.createMap()
+    val packageManager = reactContext.packageManager
+    val packageName = reactContext.packageName
+
+    val packageInfo =  packageManager.getPackageInfo(packageName, 0)
+    val applicationInfo = packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
+    val appName = packageManager.getApplicationLabel(applicationInfo).toString()
+    val runtimeVersion = getMetadataValue("expo.modules.updates.EXPO_RUNTIME_VERSION")
+    val sdkVersion = getMetadataValue("expo.modules.updates.EXPO_SDK_VERSION")
+    var appIcon = getApplicationIconUri()
+
+//    TODO - get manifest values
+
+    map.apply {
+      putString("appVersion", packageInfo.versionName)
+      putString("appName", appName)
+      putString("appIcon", appIcon)
+      putString("runtimeVersion", runtimeVersion)
+      putString("sdkVersion", sdkVersion)
+    }
+
+    promise.resolve(map)
+  }
+
+  override fun copyToClipboardAsync(content: String, promise: Promise) {
+    val clipboard = reactContext.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+    val clip = ClipData.newPlainText(null, content)
+    clipboard.setPrimaryClip(clip)
+    promise.resolve(null)
+  }
+
+  private fun getMetadataValue(key: String): String {
+    val packageManager = reactContext.packageManager
+    val packageName = reactContext.packageName
+    val applicationInfo = packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
+    var metaDataValue = ""
+
+    if (applicationInfo.metaData != null) {
+      val value = applicationInfo.metaData.get(key)
+
+      if (value != null) {
+        metaDataValue = value.toString()
+      }
+    }
+
+    return metaDataValue
```

Comment:
```suggestion
    return applicationInfo.metaData?.get(key)?.toString() ?: ""
```
It should give you the same output ;)